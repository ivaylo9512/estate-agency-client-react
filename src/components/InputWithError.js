const InputWithError = ({error, classname = '', input}) => {

    return(
        <div className={(error ? 'error ' : '') + classname}>
            {input}
            {error && 
                <div>
                    {error}
                </div>
            }
        </div>
    )
}

export default InputWithError