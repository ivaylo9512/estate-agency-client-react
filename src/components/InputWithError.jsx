import { getId } from "hooks/useInput"

const InputWithError = ({error, classname = '', input}) => {

    return(
        <div className={(error ? 'error ' : '') + classname}>
            {input}
            {error && 
                <div data-testid={`${getId(input)}Error`}>
                    {error}
                </div>
            }
        </div>
    )
}

export const getContainerId = input => `${getId(input)}Container`;

export default InputWithError