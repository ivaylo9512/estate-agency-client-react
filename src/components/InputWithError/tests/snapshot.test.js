import { shallow } from 'enzyme';
import InputWithError from '../InputWithError';

describe('InputWithError snapshot tests', () => {
    const input = <input data-testid='input-id' />

    const createWrapper = () => shallow(
        <InputWithError error='error' className='classname' input={input} />
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper();

        expect(wrapper).toMatchSnapshot();
    })
})