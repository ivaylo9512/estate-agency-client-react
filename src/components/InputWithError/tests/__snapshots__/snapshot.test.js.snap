// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithError snapshot tests should match snapshot 1`] = `
<div
  className="error "
>
  <input
    data-testid="input-id"
  />
  <div
    data-testid="input-idError"
  >
    error
  </div>
</div>
`;
